microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/properties/PropertiesFileCompletions.java

```diff
@@ -596,4 +630,29 @@ protected String getExpr(TextDocument document, int completionOffset) {
 		return snippetRegistry;
 	}
 
+	private CompletionItemDefaults getCompletionItemDefaults() {
+		if (completionItemDefaults == null) {
+			completionItemDefaults = new CompletionItemDefaults();
+		}
+		return completionItemDefaults;
+	}
+
+	private void initializeCompletionListItemDefaults(PropertiesModel model,
+			MicroProfileCompletionCapabilities completionCapabilities, int offset) {
+		CompletionItemDefaults itemDefaults = getCompletionItemDefaults();
+		boolean snippetsSupported = completionCapabilities.isCompletionSnippetsSupported();
+		if (completionCapabilities.isCompletionListItemDefaultsInsertTextFormatSupport()) {
+			itemDefaults.setInsertTextFormat(snippetsSupported ? InsertTextFormat.Snippet : InsertTextFormat.PlainText);
+		}

```

Comment:
Put these steps in `getCompletionItemDefaults()` so that you only do them once. I realize you will need to set the range every time you respond to a completion request, but once the server is initialized, the `isCompletionSnippetsSupported()` shouldn't change.